tests/integration/test_airgapped.py

```diff
@@ -0,0 +1,155 @@
+                },
+            )
+
+            logger.info(f"Waiting for {PRO_AIRGAPPED_SERVER_NAME}")
+            await ops_test.model.wait_for_idle(
+                apps=[PRO_AIRGAPPED_SERVER_NAME], status=ACTIVE_STATUS, raise_on_blocked=False, timeout=600
+            )
+
+            offer_name = f"offer-{uuid.uuid4().hex[0:4]}"
+            logger.info("Creating application offer for cross-model relation: %s", offer_name)
+            await ops_test.model.create_offer(
+                endpoint=PRO_AIRGAPPED_SERVER_ENDPOINT,
+                application_name=PRO_AIRGAPPED_SERVER_NAME,
+                offer_name=offer_name,
+            )
+
+    with ops_test.model_context(k8s_model_name):
+        async with ops_test.fast_forward():
+            logger.info("Creating SaaS")
+            # When using the `ops_test.model.consume` method like this:
+            #
+            #   await ops_test.model.consume(endpoint=f"{lxd_model_name}.{offer_name}")
+            #
+            # We get a `KeyError` saying `ops_test.model.info` has no `username`
+            # key. To make it work we have to specify the username maunually by
+            # picking one of the entries from the `ops_test.model.info.users`
+            # list, which could result in unexpected behavior if there were more
+            # than one.
+            #
+            # Another option is to use the Juju client's `consume` command which
+            # handles this by defaulting to the current user.
+            await ops_test.juju("consume", f"{lxd_model_name}.{offer_name}")
+            logger.info("Integrating Livepatch and pro-airgapped-server")
+            await ops_test.model.integrate(APP_NAME, offer_name)
+            logger.info("Waiting for Livepatch")
+            await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", raise_on_blocked=False, timeout=600)
+
+    with ops_test.model_context(lxd_model_name):
+        async with ops_test.fast_forward():
+            logger.info(f"Waiting for {PRO_AIRGAPPED_SERVER_NAME}")
+            await ops_test.model.wait_for_idle(
+                apps=[PRO_AIRGAPPED_SERVER_NAME], status=ACTIVE_STATUS, raise_on_blocked=False, timeout=600
+            )
+
+    with ops_test.model_context(k8s_model_name):
+        logger.info("Getting Livepatch model status")
+        status = await ops_test.model.get_status()  # noqa: F821
+        logger.info(f"Status: {status}")
+        assert ops_test.model.applications[APP_NAME].status == ACTIVE_STATUS
+
+        app_address = status["applications"][APP_NAME]["units"][f"{APP_NAME}/0"]["address"]
+        url = f"http://{app_address}:8080/debug/status"
+        logger.info("Querying app address: %s", url)
+        r = requests.get(url, timeout=2.0)
+        logger.info(f"Output = {r.json()}")
+        assert r.status_code == 200
```

Comment:
I would flip these two lines :) `.json()` may raise an exception if it returned text or no response.